frontend/userForm.tsx

```diff
@@ -126,11 +126,13 @@ export function UserForm({id}: {id: number | 'new' | undefined}) {
             break; // updated user
           case "success":
           case "created":
-            history.push(`/manage/user/${data?.data}`)
             unstable_batchedUpdates(() => {
               setNew(false);
+              users.push({email: user.email, id: data.data, name: user.email})
+              setUsers(Object.assign([], users))
```

Comment:
Never seen Object.assign on arrays :open_mouth: 

You can use `setUsers([...users, {email: user.email, id: data.data, name: user.email}])`. This way you don't need to mutate the array.

Maybe it's slightly clearer on two lines
```typescript
  const new_user = {email: user.email, id: data.data, name: user.email}
  setUsers([...users, new_user])
``

Is both name and email supposed to be user.email?

Reply:
yeah, there're no usernames, so I use the e-mail for both. I could set `name` to `undefined` I guess... might be clearer... And yeah, your syntax is prettier! :D